### PR TITLE
Fix Spotbugs in azure-core

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java [hunk 1]
```diff
@@ -401,4 +404,14 @@ private boolean doesArgsHaveThrowable(Object... args) {
...
     }
+
+    /**
+     * Removes CRLF pattern in the {@code logMessage}.
+     *
+     * @param logMessage The log message to sanitize.
+     * @return The updated logMessage.
+     */
+    private static String sanitizeLogMessageInput(String logMessage) {
+        return CRLF_PATTERN.matcher(logMessage).replaceAll("");
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java:415]: So, effectively we aren't allowing Windows style newlines in logging?

File: sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java [hunk 2]
```diff
@@ -35,6 +36,7 @@
  * @see Configuration
  */
 public class ClientLogger {
+    private static final Pattern CRLF_PATTERN = Pattern.compile("[\r\n]");
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java:39]: Are we looking to match `\r` or `\n` or `\r\n`? The regex is currently matching the former.

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 3]
```diff
@@ -2433,4 +2433,18 @@
...
+      <Class name="com.azure.core.util.logging.ClientLogger" />
+      <Bug pattern="CRLF_INJECTION_LOGS" />
+  </Match>
+  <Match>
+      <Class name="com.azure.core.http.policy.ExponentialBackoff" />
+      <Bug pattern="PREDICTABLE_RANDOM" />
+  </Match>
+  <Match>
+      <Class name="com.azure.core.implementation.logging.DefaultLogger" />
+      <Bug pattern="INFORMATION_EXPOSURE_THROUGH_AN_ERROR_MESSAGE" />
```
Comment (anuchandy) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml:2448]: Should we leave a comment that in the default log level we will not print the stack trace, but when the user opt-in for the lowest level log (DEBUG) we'll print it, which is expected.

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 4]
```diff
@@ -2433,4 +2433,18 @@
     <Field name="detectionConfiguration" />
     <Bug pattern="UWF_UNWRITTEN_FIELD" />
   </Match>
+  
+  <!-- Disabling false positives in azure-core -->
+  <Match>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml:2438]: Can we add a reason why these are false positives? A few months from now, if we decide to create another baseline for spotbugs, we won't remember why.

**Review by conniey — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java [hunk 5]
```diff
@@ -401,4 +404,14 @@ private boolean doesArgsHaveThrowable(Object... args) {
...
         return Arrays.copyOf(args, args.length - 1);
     }
+
+    /**
+     * Removes CRLF pattern in the {@code logMessage}.
+     *
+     * @param logMessage The log message to sanitize.
+     * @return The updated logMessage.
+     */
+    private static String sanitizeLogMessageInput(String logMessage) {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java:414]: Add unit tests to validate this.

**Review by srnagar — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java (hunk 2 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java:39]: it is either of \r or \n.
The solution is recommended here: https://find-sec-bugs.github.io/bugs.htm#CRLF_INJECTION_LOGS

File: sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java (hunk 1 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java:415]: it is either of \r or \n.
The solution is recommended here: https://find-sec-bugs.github.io/bugs.htm#CRLF_INJECTION_LOGS

**Review by alzimmermsft — APPROVED** [review]

**Comment by JonathanGiles** [comment]
This PR is lacking clarity on **why** this change was made. Without context, all I am seeing is that our logging will be slower now, and I don't know why. Future developers will not have any context, either in the code or in this PR. Please be sure to create PRs with proper descriptions of the intent of the change, and ideally include appropriate inline code comments about why things are being done the way they are.

All reviewers too - this should be baseline expectations to ensure we are not doing our future selves and teammates a disservice by under-documenting things.

**Comment by g2vinay** [comment]
> This PR is lacking clarity on **why** this change was made. Without context, all I am seeing is that our logging will be slower now, and I don't know why. Future developers will not have any context, either in the code or in this PR. Please be sure to create PRs with proper descriptions of the intent of the change, and ideally include appropriate inline code comments about why things are being done the way they are.
> 
> All reviewers too - this should be baseline expectations to ensure we are not doing our future selves and teammates a disservice by under-documenting things.

The descriptions are limited intentionally.

**Pull request merged**